client/collection_view.coffee

```diff
@@ -43,7 +43,12 @@ Template._houston_collection_view.helpers
   custom_selector_error: -> Houston._session("custom_selector_error")
   field_filter_disabled: -> if Houston._session("custom_selector") then "disabled" else ""
   headers: -> get_collection_view_fields(@name)
-  nonid_headers: -> get_collection_view_fields(@name)[1..]
+  nonid_headers: ->
+    headers = get_collection_view_fields(@name)
+    if headers[0]?._required # do not remove _id if it is required by schema
+      headers
+    else
+      headers[1..]
```

Comment:
This is the right level at which to be making this change -  it makes the code less self-documenting. 

A method called `nonid_headers` should not leave `id` in some cases. We either need to rename the method, or (more likely) remove `id` and make sure the schema doesn't complain elsewhere.